drivers/bmp180/bmp180_saul.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2016 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_bmp180
+ * @{
+ *
+ * @file
+ * @brief       SAUL adaption for BMP180 device
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "saul.h"
+#include "bmp180.h"
+#include "xtimer.h"
+
+static int32_t temperature, pressure;
+
+static int check_and_read_temperature(void *dev, phydat_t *res)
+{
+    bmp180_t *d = (bmp180_t *)dev;
+
+    bmp180_read_temperature(d, &temperature);
+    res->val[0] = temperature;
+    res->unit = UNIT_TEMP_C;
+    res->scale = -1;
+    return 1;
+}
+
+static int check_and_read_pressure(void *dev, phydat_t *res)
+{
+    bmp180_t *d = (bmp180_t *)dev;
+
+    bmp180_read_pressure(d, &pressure);
+
+    res->val[0] = pressure/10;
+    res->unit = UNIT_PA;
+    res->scale = 1;
```

Comment:
It's because I divided it by 10 so I had to shift de floating point using one to keep the right dimension.
The division by 10 is required because the range of pressure value (~100000 in Pa) of greater than the maximum of signed int (~ +-32000). The pressure output string is not super nice but correct.